src/Components/Body/MovieCard/MovieCard.tsx

```diff
@@ -1,19 +1,45 @@
 import './MovieCard.scss';
+import Popup from 'reactjs-popup';
+import {useState} from 'react';
+import {Movie} from '../../App';
+import {DeleteModal} from '../../Modals/DeleteModal/DeleteModal';
+import {EditModal} from '../../Modals/EditModal/EditModal';
 
-export interface Movie {
-    image: string,
-    title: string,
-    genre: string,
-    year: number,
+interface MovieCardProps{
+  movie: Movie,
+  onDelete: (number) => void,
 }
 
-export function MovieCard(movie: Movie) {
+export function MovieCard({movie, onDelete}: MovieCardProps) {
+  const [menuOpenState, setMenuOpen] = useState(false);
+  const [editOpen, setEditOpen] = useState(false);
+  const [deleteOpen, setDeleteOpen] = useState(false);
+
+  function onDeleteId() {
+    onDelete(movie.id);
+  }
+
   return (
     <div className='movie-card-container'>
       <img className='movie-image' src={`/static/${movie.image}`}/>
       <p className='movie-title'>{movie.title}</p>
-      <p className='movie-year'>{movie.year}</p>
+      <p className='movie-year'>{movie.date.getFullYear()}</p>
       <p className='movie-genre'>{movie.genre}</p>
+      <Popup arrow={false} trigger={
+        <button className='movie-menu-button'>
+          <div className='movie-menu-button-dot first'/>
+          <div className='movie-menu-button-dot'/>
+          <div className='movie-menu-button-dot'/>
+        </button>}
+      open={menuOpenState} onOpen={() => setMenuOpen(true)}>
+        <div className="movie-menu">
```

Comment:
I would create separate menu component and you might use common practice for menu (list tags)